ouroboros-consensus/src/Ouroboros/Consensus/Storage/ImmutableDB/Impl/Index/Primary.hs

```diff
@@ -641,7 +641,7 @@ backfillChunk chunkInfo chunk (NextRelativeSlot nextExpected) offset =
 ------------------------------------------------------------------------------}
 
 (!) :: (HasCallStack, V.Unbox a) => Vector a -> Int -> a
-v ! i
```

Comment:
Is this for `hasktags` or some such? If it's just an aesthetics change, please remove it.

Reply:
`ghcid` tripped over it

Reply:
I can put this in a separate commit; I didn't intended to commit this, but it would be quite useful for the networking team.  Working with network integration is hard because building `lib:ouroboros-consensus` after a simple change in `ouroboros-network` requires re-building > 200 modules which are independent of `ouroboros-netowrk`. `ghcid` allows to mitigate this.  We have an issue #2614 to split the consensus into smaller logical units which I would like to resurrect once we merge `p2p-master` to `master`.

Reply:
Other than that, I agree that `(!)` notation is ugly.

Reply:
The error I am getting is:
```
src/Ouroboros/Consensus/Storage/ImmutableDB/Impl/Index/Primary.hs:643:1-3: error:
    The type signature for ‘!’ lacks an accompanying binding
      (The type signature must be given where ‘!’ is declared)
    |
643 | (!) :: (HasCallStack, V.Unbox a) => Vector a -> Int -> a
    | ^^^
src/Ouroboros/Consensus/Storage/ImmutableDB/Impl/Index/Primary.hs:650:12-14: error:
    The INLINE pragma for ‘!’ lacks an accompanying binding
      (The INLINE pragma must be given where ‘!’ is declared)
    |
650 | {-# INLINE (!) #-}
    |            ^^^
```

I think that's caused by `:set -XBangPatterns` in my `~/.ghci`; I'll remove it, and clean the commit.

Reply:
Huh! I've never come across that. Seems like a GHC bug to me, on first thought -- but many GHC things require more than the first thought...

Thanks for adjusting your setup.